### PR TITLE
Add flag to allow overwriting config when the service is stopped

### PR DESCRIPTION
**Pull request by ddcprg**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -177,6 +176,19 @@ This is controlled by the property: `primary-meta-store.access-control-type` It
...
+Waggle Dance reads and writes federation configuration from and to a YAML file - refer to the section [federation](#federation) for details.
+
+The following properties are configured in the server configuration file and control the behaviour of the YAML federation storage:
+
+    yaml-storage:
+      overwrite-config-on-shutdown: true
+
+| Property                          | Required   | Description |
+|:----|:----:|:----|
+| `overwrite-config-on-shutdown`    | No         | Controls whether the federations configuration must be overwritten when the server is stopped. Settings this to `false` will cause that any federations dynamically added at runtime will be lost when the server is stopped. This is also the case of databases created at runtime when `database-resolution` is set to `MANUAL`. Default is `true`. |
```
Comment (massdosage) [inline README.md]: cause that any federations dynamically added at runtime will be lost when the server is stopped

->

cause any federations dynamically added at runtime to be lost when the server is stopped

Comment (ddcprg) [inline README.md]: ok

File: waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/FederationsAdminClient.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+import com.hotels.bdp.waggledance.api.model.AbstractMetaStore;
+
+//@Headers({ "Content-Type: application/json", "Accept: application/json" })
```
Comment (massdosage) [inline waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/FederationsAdminClient.java]: Remove comment

File: waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java [hunk 3]
```diff
@@ -413,4 +419,110 @@ private Federations stopServerAndGetConfiguration() throws Exception, FileNotFou
...
+    proxy.createDatabase(new Database("newDB", "", new File(localWarehouseUri, "newDB").toURI().toString(), null));
+    Database newDB = proxy.getDatabase("newDB");
+    assertNotNull(newDB);
+
+    Federations federations = stopServerAndGetConfiguration();
+
+    List<FederatedMetaStore> federatedMetastores = federations.getFederatedMetaStores();
+    assertThat(federatedMetastores.size(), is(1));
+
+    List<String> mappedDatabases = federatedMetastores.get(0).getMappedDatabases();
```
Comment (massdosage) [inline waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java]: DB created above will be in primary but this is checking federated

File: waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/FederationsAdminClient.java (hunk 2 above)
Comment (ddcprg) [inline waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/FederationsAdminClient.java]: done

File: waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java (hunk 3 above)
Comment (ddcprg) [inline waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java]: truth, done

**Pull request merged**